### ISSUE TITLE
Slow channel/room switching

### ISSUE THREAD

**Issue by tjconcept** [gitterHQ/gitter#562]
It's a pain to jump between multiple rooms/channels because of the loading.

Does the render have to wait for a reload? Why not just show the last know data until further updates are received? Or even better, keep each room/channel up-to-date so that switching won't even issue requests/waiting.

**Comment by trevorah** [comment]
@suprememoocow has made some performance gains around here over the past 2 weeks, but we are still hitting the network at the moment. The next step is to cache messages on the browser...

**Comment by nkbt** [comment]
Yeah, caching would be great. With pushing of new messages in background, so you can switch instantly. Even without push if you just show currently cached messages right away and then load more messages when possible - would be so much better. Non-blocking loader would be nice too. Now it is quite bad in Australia

![slow-gitter](https://cloud.githubusercontent.com/assets/175264/8185518/411c2928-1488-11e5-9f9c-42fca20b5191.gif)

**Comment by riemers** [comment]
2015, you would think things have changed by then. I am in a active channel, whenever i want to check that channel it will "catchup" all the old messages. This took last time about 5 minutes. With task manager i could see it growing in memory size (which i don't mind) why doesn't it update the client on the background? Do i miss some option or is this still an issue?
